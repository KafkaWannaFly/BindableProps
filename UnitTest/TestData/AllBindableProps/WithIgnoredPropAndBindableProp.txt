[AllBindableProps]
public partial class MyClass
{
    // Default field
    string text;

    // Support field with a default value
    string placeHolder = "Do you trust me?";

    // This field will be handled by BindableProp
    [BindableProp(
        DefaultBindingMode = (int)BindingMode.TwoWay,
        ValidateValueDelegate = nameof(ValidateValue)
        )]
    string message = "With every cell in my body!";

    [IgnoredProp]
    bool isBusy; // Don't touch!

    // If you have existing props, we don't touch them
    public static readonly BindableProperty ErrorProperty = BindableProperty.Create(
            nameof(Error),
            typeof(string),
            typeof(MyClass),
            "Things just get out of hand",
            (BindingMode)(int)BindingMode.OneWayToSource
        );

    // Also not touch this prop
    public string Error
    {
        get => (string)GetValue(TextInput.ErrorProperty);
        set
        {
            SetValue(MyClass.ErrorProperty, value);
        }
    }

    static bool ValidateValue(BindableObject bindable, object value)
    {
        return true;
    }
    
    public TextInput()
    {
        InitializeComponent();
    }
}