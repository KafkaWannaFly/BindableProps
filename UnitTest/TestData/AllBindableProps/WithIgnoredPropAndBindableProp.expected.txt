// <auto-generated/>

namespace global
{
    public partial class MyClass
    {

        public static readonly BindableProperty TextProperty = BindableProperty.Create(
                    nameof(Text),
                    typeof(string),
                    typeof(MyClass),
                    default
                );

        public string Text
        {
            get => text;
            set 
            { 
                text = value;
                SetValue(MyClass.TextProperty, text);
            }
        }

        public static readonly BindableProperty PlaceHolderProperty = BindableProperty.Create(
                    nameof(PlaceHolder),
                    typeof(string),
                    typeof(MyClass),
                    "Do you trust me?"
                );

        public string PlaceHolder
        {
            get => placeHolder;
            set 
            { 
                placeHolder = value;
                SetValue(MyClass.PlaceHolderProperty, placeHolder);
            }
        }

    }
}